soroban-sdk-macros/src/lib.rs

```diff
@@ -119,7 +119,13 @@ pub fn contracttype(metadata: TokenStream, input: TokenStream) -> TokenStream {
     };
     let input = parse_macro_input!(input as DeriveInput);
     let ident = &input.ident;
-    let gen_spec = matches!(input.vis, Visibility::Public(_));
+    // If the export argument has a value, do as it instructs regarding
+    // exporting. If it does not have a value, export if the type is pub.
+    let gen_spec = if let Some(export) = args.export {
+        export
+    } else {
+        matches!(input.vis, Visibility::Public(_))
+    };
```

Comment:
I really wanted to write a test for this, but it is rather difficult without setting up a full crate that builds a wasm then analyzes its contract spec, which is not efficient. I think we will need to catch issues with this code via examples and test vectors, which we do have in other repos. It isn't ideal.